packages/compiler/src/module-resolver/module-resolver.ts

```diff
@@ -162,11 +162,8 @@ export async function resolveModule(
       const pkgFile = resolvePath(dir, "package.json");
       if (!(await isFile(host, pkgFile))) continue;
       const pkg = await readPackage(host, pkgFile);
-      if (pkg.name === name) {
-        return loadPackage(dir, pkg);
-      } else {
-        return undefined;
-      }
+      if (pkg.name !== name) continue;
+      return loadPackage(dir, pkg);
```

Comment:
I think the way it is done now is the correct way according the the node resolution spec, you only lookup to the first parent package.json for resolving self. What is the use case?

Reply:
The case happened in our cadl-ranch test where during generation our emitter dist code is in the same folder with tsp code and emitter is not released yet.